includes/Modules/Analytics.php

```diff
@@ -1395,32 +1367,29 @@ protected function validate_report_metrics( $metrics ) {
 			)
 		);
 
-		$invalid_metrics = array_diff( $metrics, $valid_metrics );
+		$invalid_metrics = array_diff(
+			array_map(
+				function ( $metric ) {
+					return $metric->getExpression();
+				},
+				$metrics
+			),
+			$valid_metrics
+		);
 
 		if ( count( $invalid_metrics ) > 0 ) {
-			return sprintf(
-				/* translators: %s is replaced with a comma separated list of the invalid metrics. */
-				_n(
-					'Unsupported metric requested: %s',
-					'Unsupported metrics requested: %s',
-					count( $invalid_metrics ),
-					'google-site-kit'
-				),
-				implode( ', ', $invalid_metrics )
-			);
+			throw new Invalid_Metrics_Exception( '', 0, null, $invalid_metrics );
```

Comment:
As mentioned before, let's give the message to the exception rather than have it generate its own. This makes the instantiation much simpler as well as we don't need to provide it with the metrics (although maybe this would make sense to do in the future), it's not needed yet.